admin/python/lsst/qserv/admin/dataLoader.py

```diff
@@ -961,47 +796,45 @@ def _makeIndexMultiNode(self, database, table, metaTable, idxCol):
         may need special optimization or parameters.
         """
 
-        czarCursor = self.mysql.cursor()
-
         # load data from all chunks
         for chunk in self.chunks:
 
             # get worker name for this chunk
             wname = self.chunkMap.worker(chunk)
+            wmgr = self.workerWmgrMap[wname]
 
-            conn = self.workerConnMap[wname]
-            wCursor = conn.cursor()
-
-            q = "SELECT {0}, chunkId, subChunkId FROM {1}.{2}_{3}"
-            q = q.format(idxCol, database, table, chunk)
-            self._log.debug('query: %s', q)
-            wCursor.execute(q)
+            # get index data
+            columns = (idxCol, 'chunkId', 'subChunkId')
+            indexData = wmgr.getIndex(database, table, chunkId=chunk, columns=columns)
 
-            q = "INSERT INTO {0} VALUES (%s, %s, %s)".format(metaTable)
-            while True:
-                seq = wCursor.fetchmany(1000000)
-                if not seq:
-                    break
-                czarCursor.executemany(q, seq)
+            # dump it into a in-memory file
```

Comment:
Yes, there is a potential problem. But it is in the memory already, hope is that data size is not too large to make another copy formatted differently.